### DISCUSSION TITLE
ThoughtSpot 8.8 no outside internet connection

### DISCUSSION THREAD

**Discussion by tb-all** [thoughtspot/cs_tools#86] [General]
We want to enable CS Tools on our recently updated 8.8 ThoughtSpot Software env - I've run python -m sysconfig - what info do you need to help us progress this? Thanks

**Comment by boonhapus** [comment]
@tb-all can you share the output of that command? It will look something similar to this post: #77

**Comment by tb-all** [comment]
[python_-m_sysconfig.txt](https://github.com/thoughtspot/cs_tools/files/13477850/python_-m_sysconfig.txt)

Here you go

**Reply by boonhapus** [reply]
@tb-all ah this is on your ThoughtSpot cluster itself. Can you try it with `python3 -m sysconfig` instead and share the output?

**Comment by tb-all** [comment]
[python3_-m_sysconfig.txt](https://github.com/thoughtspot/cs_tools/files/13487069/python3_-m_sysconfig.txt)

Here you go. Thanks

**Reply by boonhapus** [reply]
Thanks @tb-all , here's an egnyte link.

```
cs-tools_1.4.13_linux-x86_64_3.9.zip
Password: 7Ld85By7Km
```
[https://thoughtspot.egnyte.com/dl/3oNdFNveSs/cs-tools_1.4.13_linux-x86_64_3.9.zip_](https://thoughtspot.egnyte.com/dl/3oNdFNveSs/cs-tools_1.4.13_linux-x86_64_3.9.zip_)

**Reply by tb-all** [reply]
Thanks - Do I just run the bootstrapped on our AWS server?

**Reply by boonhapus** [reply]
Apologies @tb-all , I just saw this one!

---

1. Extract the zip file.
2. Run python against the `_bootstrapper.py` file with `--offline-mode` specified

`cd cs-tools_1.4.13_linux-x86_64_3.9.zip`
`python3 _bootstrapper.py --install --offline-mode .`

**Reply by tb-all** [reply]
Thanks....

The install did start but I got this....
15:16:23 | Welcome to the CS Tools Bootstrapper!
         | Ideally, you will only need to run this one time, and then your environment can be fully managed by CS Tools itself.
         | If you run into any issues, please reach out to us on GitHub Discussions below.
         |
         |           GitHub: https://github.com/thoughtspot/cs_tools/issues/new/choose
         |
         | Creating the CS Tools virtual environment.
15:16:27 | Looking in links: .
         | Processing ./pip-23.3.1-py3-none-any.whl
         | Installing collected packages: pip
         | Attempting uninstall: pip
         | Found existing installation: pip 21.2.4
         | Uninstalling pip-21.2.4:
         | Successfully uninstalled pip-21.2.4
15:16:29 | Successfully installed pip-23.3.1
         | Using the offline binary found at .
         | Installing CS Tools and its dependencies.
         | Looking in links: .
         | Processing ./cs_tools-1.4.13.zip
15:16:30 | Installing build dependencies: started
15:16:31 | Installing build dependencies: finished with status 'done'
         | Getting requirements to build wheel: started
         | Getting requirements to build wheel: finished with status 'done'
         | Installing backend dependencies: started
15:16:32 | Installing backend dependencies: finished with status 'done'
         | Preparing metadata (pyproject.toml): started
         | Preparing metadata (pyproject.toml): finished with status 'done'
         | Requirement already satisfied: pip>=23.1 in /home/admin/.config/cs_tools/.cs_tools/lib/python3.9/site-packages (from cs_tools[cli]) (23.3.1)
         | Processing ./thoughtspot_tml-2.0.10-py3-none-any.whl (from cs_tools[cli])
         | Processing ./awesomeversion-23.11.0-py3-none-any.whl (from cs_tools[cli])
         | Processing ./httpx-0.25.2-py3-none-any.whl (from cs_tools[cli])
         | Processing ./pydantic-1.10.13-py3-none-any.whl (from cs_tools[cli])
         | Processing ./pendulum-2.1.2.tar.gz (from cs_tools[cli])
15:16:33 | Installing build dependencies: started
         | Installing build dependencies: finished with status 'done'
         | Getting requirements to build wheel: started
15:16:34 | Getting requirements to build wheel: finished with status 'done'
         | Preparing metadata (pyproject.toml): started
         | Preparing metadata (pyproject.toml): finished with status 'done'
         | Processing ./rich-13.7.0-py3-none-any.whl (from cs_tools[cli])
         | Processing ./sqlmodel-0.0.12-py3-none-any.whl (from cs_tools[cli])
         | Processing ./toml-0.10.2-py2.py3-none-any.whl (from cs_tools[cli])
         | Processing ./StrEnum-0.4.15-py3-none-any.whl (from cs_tools[cli])
         | Processing ./typer-0.9.0-py3-none-any.whl (from typer[all]>=0.9.0; extra == "cli"->cs_tools[cli])
         | Processing ./uvicorn-0.24.0.post1-py3-none-any.whl (from cs_tools[cli])
         | Processing ./fastapi-0.104.1-py3-none-any.whl (from cs_tools[cli])
         | Processing ./anyio-3.7.1-py3-none-any.whl (from fastapi>=0.89.1->cs_tools[cli])
         | Processing ./starlette-0.27.0-py3-none-any.whl (from fastapi>=0.89.1->cs_tools[cli])
         | Processing ./typing_extensions-4.8.0-py3-none-any.whl (from fastapi>=0.89.1->cs_tools[cli])
         | Processing ./certifi-2023.11.17-py3-none-any.whl (from httpx>=0.24.0->cs_tools[cli])
         | Processing ./httpcore-1.0.2-py3-none-any.whl (from httpx>=0.24.0->cs_tools[cli])
         | Processing ./idna-3.6-py3-none-any.whl (from httpx>=0.24.0->cs_tools[cli])
         | Processing ./sniffio-1.3.0-py3-none-any.whl (from httpx>=0.24.0->cs_tools[cli])
         | Processing ./h11-0.14.0-py3-none-any.whl (from httpcore==1.*->httpx>=0.24.0->cs_tools[cli])
         | Processing ./python_dateutil-2.8.2-py2.py3-none-any.whl (from pendulum>=2.1.2->cs_tools[cli])
         | Processing ./pytzdata-2020.1-py2.py3-none-any.whl (from pendulum>=2.1.2->cs_tools[cli])
         | Processing ./markdown_it_py-3.0.0-py3-none-any.whl (from rich>=13.0.0->cs_tools[cli])
         | Processing ./pygments-2.17.2-py3-none-any.whl (from rich>=13.0.0->cs_tools[cli])
         | Processing ./SQLAlchemy-2.0.23-py3-none-any.whl (from sqlmodel>=0.0.8->cs_tools[cli])
         | Processing ./PyYAML-6.0.1.tar.gz (from thoughtspot-tml>=2.0.6->cs_tools[cli])
         | Installing build dependencies: started
15:16:35 | Installing build dependencies: finished with status 'error'
         | error: subprocess-exited-with-error
         | × pip subprocess to install build dependencies did not run successfully.
         | │ exit code: 1
         | ╰─> [5 lines of output]
         | Looking in links: .
         | Processing ./setuptools-69.0.2-py3-none-any.whl
         | Processing ./wheel-0.42.0-py3-none-any.whl
         | Could not find a version that satisfies the requirement Cython<3.0 (from versions: none)
         | No matching distribution found for Cython<3.0
         | [end of output]
         | note: This error originates from a subprocess, and is likely not a problem with pip.
         | error: subprocess-exited-with-error
         | × pip subprocess to install build dependencies did not run successfully.
         | │ exit code: 1
         | ╰─> See above for output.
         | note: This error originates from a subprocess, and is likely not a problem with pip.
         | Unexpected error in bootstrapper, see /home/admin/cs-tools_1.4.13_linux-x86_64_3.9/cs_tools-bootstrap-error-tmpdq1lil3o.log for details..

**Reply by boonhapus** [reply]
Thanks @tb-all . I believe this is likely because the AWS build don't include python development libraries and the ssl library by default. Can you share the log file that written at `/home/admin/cs-tools_1.4.13_linux-x86_64_3.9/cs_tools-bootstrap-error-tmpdq1lil3o.log` ?

**Comment by tb-all** [comment]
[cs_tools-bootstrap-error-tmpdq1lil3o.log](https://github.com/thoughtspot/cs_tools/files/13557848/cs_tools-bootstrap-error-tmpdq1lil3o.log)

**Reply by tb-all** [reply]
Here you go

**Reply by boonhapus** [reply]
@tb-all Hmm, it looks like an older version of CS Tools may been installed on this cluster? (There's reference to `thoughtspot_tml==2.0.6`.) Can we try it with the `--reinstall` option instead?

```
cd cs-tools_1.4.13_linux-x86_64_3.9.zip
python3 _bootstrapper.py --reinstall --offline-mode .
```

Should that not work either, I'd recommend get a call together so I can debug live. I'm in US/Central time, but can be flexible to help unblock you.

**Reply by tb-all** [reply]
Thanks...this failed again. 
[cs_tools-bootstrap-error-tmpyzu6nm1x.log](https://github.com/thoughtspot/cs_tools/files/13581585/cs_tools-bootstrap-error-tmpyzu6nm1x.log)

**Reply by tb-all** [reply]
Shall I raise a support ticket?

**Reply by boonhapus** [reply]
@tb-all Support will be unable to help. Can you shoot me an email so we can coordinate some time to meet?

nicholas.cooper@thoughtspot.com

**Reply by boonhapus** [reply]
@tb-all Looking into this more ... the problem appears to be **[pyyaml#736](https://github.com/yaml/pyyaml/issues/736)** / **[v6.0.1 compatability](https://github.com/yaml/pyyaml/commit/c42fa3bff1eabdb64763bb1526d9ea1ccb708479)** on Linux builds. We use `pyyaml >= 6.0` in the upstream **[`thoughtspot_tml` library](https://github.com/thoughtspot/thoughtspot_tml/blob/a9d6715c3e9a1b345121dbbbae91a202a544a763/pyproject.toml#L34C21-L34C21)** and I'm not interested in supporting `5.4.1` (the previous major version).

I'll have to explore ways to get this to work on the cluster / Linux systems. I **_think_** I should be able to build the wheel programmatically and supply it, but need to test this. I'll try this today.
